internal/provider/resource_consul_cluster.go

```diff
@@ -207,6 +208,28 @@ func resourceConsulCluster() *schema.Resource {
 				Type:        schema.TypeString,
 				Computed:    true,
 			},
+			"ip_allowlist": {
```

Comment:
New `ip_allowlist` field schema.

`schema.TypeList` does not support a `ValidateFunc`, so I added them under the individual fields `description` and `address`

This also means I had to create a separate validate function to verify duplicates. See `buildIPAllowlist`.

If we went with `schema.TypeSet` to hold the collection, we wouldn't need to validate duplicates. However, when I tested that locally, the behaviour will simply remove duplicates and use a single one. There is no message to the end user, which doesn't seem like the right user experience we want. (Trade-off:  more code vs better ux)

Reply:
Another option to remove the `buildIPAllowlist` function, is to not do duplicate validation at the terraform provider layer, since the backend does it anyways.

Reply:
thanks, updated to remove that logic 👍🏽 will resolve this comment